packages/core/types/src/dml/index.ts

```diff
@@ -163,3 +163,10 @@ export type InferTypeOf<T extends IDmlEntity<any>> = InstanceType<Infer<T>>
 export type InferEntityType<T extends any> = T extends IDmlEntity<any>
   ? InferTypeOf<T>
   : T
+
+export type EntityIndex = {
+  name?: string
+  unique?: boolean
+  fields: string[]
```

Comment:
thought: I suggest going back to using `on` instead of `fields`. According to the DML terminology, we refer to the items in a schema as properties, not fields. I think we should be consistent with that.

Reply:
Agree on consolidating the terminology. 💯 Is there a docu on this yet that I can read?

From what I see on the codebase, we have this:

```
define('modelName', {
  [field]: model.text() -> property
})
```

We seem to define the key of the schema as a field and the value of the schema field a property.

Reply:
I read it in the Notion document that Virk wrote.

The field/property detail from the codebase is interesting, but ultimately, the developer doesn't need to know its details. If we call it a property/field, then it's best to stick with the choice.

Reply:
@shahednasser should we switch to on?

Reply:
I think that's better

Reply:
done! 🥇